pandas/core/series.py

```diff
@@ -273,6 +273,25 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
     @classmethod
     def from_array(cls, arr, index=None, name=None, dtype=None, copy=False,
                    fastpath=False):
+        """
+        DEPRECATED: use the pd.Series(..) constructor instead.
+
+        """
+        warnings.warn("'from_array' is deprecated and will be removed in a "
+                      "future version. Please use the pd.Series(..) "
+                      "constructor instead.", FutureWarning, stacklevel=2)
+        cls._from_array(arr, index=index, name=name, dtype=dtype, copy=copy,
+                       fastpath=fastpath)
+
+    @classmethod
+    def _from_array(cls, arr, index=None, name=None, dtype=None, copy=False,
```

Comment:
its prob easier just to elminate this and deal with SparseSeries directly in the Series constructor. Do we have an issue for this?

Reply:
annoyhing we'd actually have to do this in ``__new__`` (but not that big of a deal)

Reply:
Yes, that would be cleaner. But it is also an API change that `Series(..)` would no longer always return a Series but depending on the input also SparseSeries. 
So the question is if we actually want this. I can open an issue about it though.